### PR TITLE
use two-stage building for conductor:server

### PR DESCRIPTION
**Pull request by houz42**
With two-stage building, we do not need a separate 'builder' image anymore. It will make the dockerfile   easy to read and maintain, and keep the resulting image 'optimized'.

### CODE REVIEW

**Comment by cyzhao** [comment]
@houz42 please explain why these changes are needed. Please always provide a description in the PR moving forward.

**Comment by houz42** [comment]
Sorry for the late response

File: docker/server/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+FROM openjdk:8-jdk AS builder
+
+MAINTAINER Netflix OSS <conductor@netflix.com>
+
+# Copy the project directly onto the image
+COPY . /conductor
+WORKDIR /conductor
+
+# Build the server on run
+ENTRYPOINT ./gradlew build -x test
```
Comment (cyzhao) [inline docker/server/Dockerfile]: Should be `RUN ./gradlew build -x test`

File: docker/server/Dockerfile [hunk 2]
```diff
@@ -1,3 +1,17 @@
+#
+# conductor:server - Netflix conductor server
+#
```
Comment (cyzhao) [inline docker/server/Dockerfile]: Please update the comment to explain this stage.

Comment (houz42) [reply docker/server/Dockerfile]: How about comment like this:
> 0. builder stage
> 1. bin stage

**Comment by houz42** [comment]
@cyzhao how can I fix the coverage error?

File: docker/server/Dockerfile (hunk 2 above)
Comment (cyzhao) [reply docker/server/Dockerfile]: Sounds good

**Comment by cyzhao** [comment]
> @cyzhao how can I fix the coverage error?

Don't worry about it given your PR doesn't touch any Java code.

**Pull request merged**